base/vulkanexamplebase.cpp

```diff
@@ -98,6 +98,22 @@ VkResult VulkanExampleBase::createInstance(bool enableValidation)
 	instanceCreateInfo.pNext = NULL;
 	instanceCreateInfo.pApplicationInfo = &appInfo;
 
+
+
+	VkDebugUtilsMessengerCreateInfoEXT debugUtilsMessengerCI{};
```

Comment:
Would it be possible to somehow merge this with the code from `setupDebugging` in `VulkanDebug.h`, which already sets up the debug messenger? Otherwise this results in code duplication. I'd also remove some verbosity flags and message types like in `setupDebugging`.

Reply:
Technically I could create a `vks::debug::setupInstanceCreateDebugging(VkInstanceCreateInfo& instanceCreateInfo)` function, but the `VkDebugUtilsMessengerCreateInfoEXT` that goes into the `instanceCreateInfo.pNext`  value needs to live beyond the scope of such a function.  it has to exist until vkCreateIntance is called and returns.  I suppose I could declare it as static isnide the `VulkanDebug.cpp` unit, but that seems weird to me.

I've tried to strike a balance with my new commit where the `VkDebugUtilsMessengerCreateInfoEXT` is still declared in `vulkanexamplebase`, but all the boilerplate for populating it is moved to a function in `VulkanDebug.cpp` that is also then used by the existing code to setup debugging.

Hope this is satisfactory, but I can switch to a static declaration if you'd prefer.